odoo/orm/environments.py

```diff
@@ -84,6 +84,12 @@ def __new__(cls, cr, uid, context, su=False):
             transaction.default_env = self
         return self
 
+    def __setattr__(self, name: str, value: typing.Any) -> None:
+        # when initialized, make attributes read-only
+        if name in vars(self):
+            raise AttributeError(f"Attribute {name!r} is read-only")
```

Comment:
It might be a good idea to give the right way to do (`env()`) for the dev experience.